src/v2/coins/abstractUtxoCoin.js

```diff
@@ -590,16 +619,15 @@ class AbstractUtxoCoin extends BaseCoin {
 
     addressDetails.coinSpecific.redeemScript = inputScript.toString('hex');
 
-    if (['p2sh-p2wsh', 'p2wsh'].includes(addressType)) {
+    if ([Codes.UnspentTypeTcomb('p2shP2wsh'), Codes.UnspentTypeTcomb('p2wsh')].includes(addressType)) {
+      addressDetails.coinSpecific.witnessScript = inputScript.toString('hex');
       const witnessScriptHash = bitcoin.crypto.sha256(inputScript);
       const redeemScript = bitcoin.script.witnessScriptHash.output.encode(witnessScriptHash);
-      const redeemScriptHash = bitcoin.crypto.hash160(redeemScript);
-      outputScript = bitcoin.script.scriptHash.output.encode(redeemScriptHash);
-      addressDetails.coinSpecific.witnessScript = inputScript.toString('hex');
-      addressDetails.coinSpecific.redeemScript = redeemScript.toString('hex');
-      if (addressType === 'p2wsh') {
-        outputScript = redeemScript;
-        delete addressDetails.coinSpecific.redeemScript;
+      outputScript = redeemScript;
```

Comment:
I don't like that we set an incorrect `outputScript` for wrapped segwit first and then overwrite it later. Rather, I would prefer if the `outputScript` would be set only explicitly for the `addressType` that it will finally apply to. Could we please reorder these lines such that there is no overwriting and explicit disambiguation of the two cases?

Reply:
Done